File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.java
class Solution {
    public int characterReplacement(String s, int k) {
        int n=s.length();
       Map<Character,Integer> mp=new HashMap<>();
       int count=0;
       int res=0;
       int start=0;
       int index=0;
       for(int i=0;i<n;i++)
       {
          mp.put(s.charAt(i),mp.getOrDefault(s.charAt(i),0)+1);
          if(mp.get(s.charAt(i))>=count)
          {
            index=i;
            count=mp.get(s.charAt(i));
          }

          if(count+k<i-start+1)
          {
            mp.put(s.charAt(start),mp.get(s.charAt(start))-1);
            if(s.charAt(index)==s.charAt(start)) count--;
            start++;
          }
          else
            res=Math.max(res,i-start+1);
       }
       return res;
        
    }
}

File: Data Structures & Algorithms/partition-labels/submission-0.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:

        arr = [0]*26

        for i,c in enumerate(s):
            arr[ord(c) - ord('a')] = i
        # x - 3, y - 4, z -7,b-9,i-10,s-11,l-12
        end = 0
        start = 0
        res = [] 
        for i in range(len(s)):
            c = s[i]
            if i > end:
                res.append(end-start+1)
                end = arr[ord(c)-ord('a')]
                start = i
            
            else:
                end = max(end,arr[ord(c)-ord('a')])
        res.append(end-start+1)
        return res        